tests/unit/helpers.py

```diff
@@ -0,0 +1,3 @@
+def get_relation_data(relations, endpoint, key):
+    """Retrieve the value for a given key from the local_app_data of a relation with the specified endpoint."""
+    return next((r.local_app_data[key] for r in relations if r.endpoint == endpoint), None)
```

Comment:
To avoid surprises in the future, we could address a potential future case of more than one relation of the same type by adding an assertion.


```suggestion
    relevant = [r.local_app_data[key] for r in relations if r.endpoint == endpoint]
    assert len(relevant) < 2, "This helper currently assumes only one relation."
    return relevant[0] if relevant else None
```